R/utils.R

```diff
@@ -181,3 +181,28 @@ create_chunk <- function(x, transform) {
 
   cat(knitr::knit(text = knitr::knit_expand(text = new_chunk), quiet = TRUE))
 }
+
+
+#' Check package versions for docorator object are the same as loaded
+#'
+#' @param x docorator object
+#' @keywords internal
+check_pkg_version <- function(x) {
+
+  if (rlang::inherits_any(x$display, c("gt_tbl", "gt_group"))) {
+    pkg <- "gt"
+  }else if (inherits(x$display,"ggplot")) {
+    pkg <- "ggplot"
+  }else{
+    return()
+  }
+
+ package_list <- append(x$session_info$otherPkgs,x$session_info$loadedOnly)
+ pkg_version <- package_list[[pkg]][["Version"]]
+
+ # get current version if it exists
+ current_version = as.character(utils::packageVersion(pkg))
+
+ if(utils::compareVersion(current_version, pkg_version)!= 0){
+   cli::cli_text("Note: docorator object was created with {.pkg {pkg} {pkg_version}}. You are now running {.pkg {pkg} {current_version}}. There may be issues rendering your document.")}
```

Comment:
hate to be pedantic, but was thinking this would be different classes - {.pkg {pkg}} {.val {pkg_version}}

Reply:
I didn't like the quotes 
<img width="434" height="51" alt="image" src="https://github.com/user-attachments/assets/7dca2e06-412d-4ec1-b397-657116e200ed" />

and preferred it like this 
<img width="206" height="55" alt="image" src="https://github.com/user-attachments/assets/e6968513-5c3c-42ff-ae4b-663661d94fcd" />